pkg/tools/etcd_helper.go

```diff
@@ -164,6 +164,10 @@ func (h *EtcdHelper) ExtractObjToList(key string, listObj runtime.Object) error
 		return err
 	}
 
+	if response == nil {
```

Comment:
Can you put this inside if err != nil, and have if IsEtcdNotFound(err) return nil?  Would be clearer.